server/src/main/java/io/druid/curator/announcement/Announcer.java

```diff
@@ -62,7 +62,7 @@
   private final CuratorFramework curator;
   private final PathChildrenCacheFactory factory;
 
-  private final List<Pair<String, byte[]>> toAnnounce = Lists.newArrayList();
+  private final List<Announceable> toAnnounce = Lists.newArrayList();
   private final List<Pair<String, byte[]>> toUpdate = Lists.newArrayList();
```

Comment:
It could although the removeParentsIfCreated part isn't relevant. I can just set it to false though.